### ISSUE TITLE
[1.16.4] Magma anvil still not working

### ISSUE THREAD

**Issue by TheSaucyWelshman** [PrinceOfAmber/Cyclic#1649]
Minecraft Version: 1.16.4

Forge Version: 35.1.13

Mod Version: 1.0.0

Single Player or Server: SSP

Describe problem (what you were doing; what happened; what should have happened):
Using the 1.1.1 version of the DW20 1.16 pack with a few additions. I updated Cyclic when I saw the closed issue about this but it still doesn't work for me.
Magma anvil still won't accept lava from pipes or buckets.

Log file link:
https://hastebin.com/anufisucoc.sql

I tried both an existing world and a fresh one for this log. Mekanism pipes in my existing world and both Mekanism and Cyclic in the fresh creative world. Nothing moved the lava into the anvil.

Video/images/gifs (direct upload or link): 

https://imgur.com/a/O68bpOy

**Comment by trunksbomb** [comment]
The Magma Anvil does not take lava, it takes Magma which is a fluid added by Cyclic and made in the Melting Chamber with two Magma Creams per 200 mB.

![image](https://user-images.githubusercontent.com/1904331/103324742-c896e400-4a16-11eb-9dde-6abce546e62e.png)

There is a separate bug, however, where the fluid level doesn't update visually unless you quit/reload or leave the area and come back.

**Comment by TheSaucyWelshman** [comment]
Huh, could have sworn the book said lava. My bad for not reading it well enough

**Comment by trunksbomb** [comment]
I double checked because that's definitely something I would have messed up lol. It's there.

![image](https://user-images.githubusercontent.com/1904331/103325465-f9c4e380-4a19-11eb-9e4d-07f3c08ea9e4.png)

**Comment by Lothrazar** [comment]
Seems to be more or less working in 1.0.1  :) . no yeah i get why its confusing. in 1.12 it took lava for sure
![1649](https://user-images.githubusercontent.com/621529/103415380-b4e29f00-4b36-11eb-9137-e26c45149a7a.gif)

**Comment by trunksbomb** [comment]
@Lothrazar In dev, the Magma Anvil was not updating its fluid level after piping some in. I had to quit/reload for it to update which means it's not getting sent to the client properly. This is because the Magma Anvil did not override `setFluid` which is called in the packet that updates client-side fluids.

**Comment by MichaelNZ404** [comment]
Works first try for me

https://user-images.githubusercontent.com/4734163/104561139-a3e77100-5692-11eb-8d04-b35f945b6d4b.mp4

**Comment by Rectorian** [comment]
Hi,
I'm encountering the same issue.

FTB Presents Direwolf20 1.16
Forge Version 35.1.36
Cyclic Version 1.1.3

I was trying to move magma into the magma anvil from the melting chamber. I am using refined pipes to move the liquid, both basic extractor and basic pipe. The magma melts, both magma cream and magma blocks, and transfers into the pipe and then seems to disappear into the magma anvil. Putting an item to repair in the repair slot does not repair at all, and I've check on both redstone sensitive and ignored modes. If I save the game and reload the world, the magma will appear inside the anvil and I am able to repair items until the magma runs out. Trying to refill with magma inside the anvil does nothing as the fluid level won't change until another reload of the save.

Here is the Latest log file:
[latest.log](https://github.com/Lothrazar/Cyclic/files/5831533/latest.log)

Extra Notes:
I am running the FTB Presents Direwolf20 1.16 pack with my own addition of apotheosis and optifine. I can check to see if either of those are interfering, but that will have to wait a couple hours until I am not busy. I also open the world to LAN sometimes and am planning to test if that seems to affect anything either.

**Comment by cornytrace** [comment]
I also have the same issue.
For me it is just a cosmetic error, as after saving/quitting the correct magma level is displayed in the anvil and items are being repaired as they should even before saving/quitting.

Modpack: All the mods 6 1.4.0
Forge Version 35.1.36
Cyclic Version 1.1.2

**Comment by Rectorian** [comment]
Update:
After reading the above comment by cornytrace, I double checked and noticed that they are correct, it seems to be just a cosmetic problem. I was able to see that my items were being repaired but that the tank visual didn't seem to change when refilled with magma.
I also checked if optifine or apotheosis were causing the issue, and after removing and checking if it seemed to be the cause of the problem, it still persisted with no change in the visual.
Hope this info can help!
